pkg/dockerRegistry/DockerRegistryIpsConfigService.go

```diff
@@ -154,6 +138,40 @@ func (impl DockerRegistryIpsConfigServiceImpl) HandleImagePullSecretOnApplicatio
 	return updatedValuesFileContent, nil
 }
 
+func (impl DockerRegistryIpsConfigServiceImpl) getDockerRegistryIdForCiPipeline(ciPipelineId int) (*string, error) {
+	ciPipeline, err := impl.ciPipelineRepository.FindById(ciPipelineId)
+	if err != nil {
+		impl.logger.Errorw("error in fetching ciPipeline", "ciPipelineId", ciPipelineId, "error", err)
+		return nil, err
+	}
+
+	if ciPipeline.IsExternal && ciPipeline.ParentCiPipeline == 0 {
+		impl.logger.Warn("Ignoring for external ci")
+		return nil, nil
+	}
+
+	if ciPipeline.CiTemplate == nil {
+		impl.logger.Warn("returning as ciPipeline.CiTemplate is found nil")
+		return nil, nil
+	}
+
+	dockerRegistryId := ciPipeline.CiTemplate.DockerRegistryId
+	if len(*dockerRegistryId) == 0 {
+		impl.logger.Warn("returning as dockerRegistryId is found empty")
+		return nil, nil
```

Comment:
throw custom error

Reply:
This custom error was not being thrown earlier also, so don't know it may impact other flows also outside this function which will increase the scope of this maybe